Fix on demand default arrays + support for Node 4.5

chfritz opened the pull request:
1. Fixes an issue with arrays whose value is not specified during message creation, when using the onTheFly message definitions.

2. Counterpart to https://github.com/RethinkRobotics-opensource/ros_msg_utils/pull/6. Once the other PR is merged in ros_msg_utils, we can change the dependency back to the RethinkRobotics-opensource repo.

chfritz committed:
Fixes arrays for complex subtypes

- for on-demand messages

chfritz committed:
- Added babel prepublish script to support Node 4.5

- Removed no longer used function.

- Added test cases for on-the-fly message definitions

chris-smith commented:
Ah! Sorry I broke the default arrays - comments for babel here will be the same as those over on RethinkRobotics-opensource/ros_msg_utils#6

chfritz commented:
You didn't break anything. They were broken all along ;-)

chfritz committed:
Fixed a bug in fixed-size array serialization

- for fixed size arrays we should *not* write the length of the array
  before the data (nor expect it when deserializing)

chris-smith commented on package.json:
```diff
@@ -24,12 +26,14 @@
     "mocha": "^2.4.5"
   },
   "dependencies": {
+    "babel-cli": "^6.18.0",
```
should these be `devDependencies`?

chfritz commented:
fixed. thanks.

chris-smith commented on package.json:
```diff
@@ -21,15 +23,17 @@
     "md5": "2.1.0",
     "moment": "2.12.0",
-    "ros_msg_utils": "RethinkRobotics-opensource/ros_msg_utils#v1.0.1",
+    "ros_msg_utils": "chfritz/ros_msg_utils#add_babel",
```
`ros_msg_utils` has been published so you should be able to set this to 1.0.2 - I think that's the only thing left to update

chfritz committed:
Using published version 1.0.2 of ros_msg_utils

chfritz commented:
great! done.
